### PR TITLE
Removes dependency on deprecated torchtext (refs #829)

### PR DESCRIPTION
**Pull request by cwmeijer**
See #829.
This PR gets rid of:
- [x] uses of `torchtext.get_tokenizer`
- [x] uses of `torchtext.vocab.Vector`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -69,11 +66,10 @@ unfixable = []
 
 per-file-ignores = {}
 
-
 # Allow unused variables when underscore-prefixed.
 dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
 
-[tool.ruff.lint.isort]
```
Comment (loostrum) [inline pyproject.toml:76]: I'm assuming this is because of some linter deprecation/error?

**Review by loostrum — CHANGES_REQUESTED** [review]
The changes look good! Could you add pandas as a dependency to setup.cfg? Then it's good to go.

File: pyproject.toml (hunk 1 above)
Comment (cwmeijer) [reply pyproject.toml:76]: indeed. Apparently this changed in the last half year and my precommit wouldn't even let me commit anything without the change ;-)

**Comment by cwmeijer** [comment]
> The changes look good! Could you add pandas as a dependency to setup.cfg? Then it's good to go.

I added the dependency already for notebooks, but it was necessary for the dashboard and unit tests, too. I think strictly it shouldn't be necessary for DIANNA base install, but I'm not a 100% sure. As it is necessary for any other type of install, I made pandas a dependency for dianna base.

**Review by loostrum — APPROVED** [review]

**Pull request merged**